### PR TITLE
Contribution policy

### PR DESCRIPTION
**Pull request by baentsch**
Tracking in Draft mode https://github.com/open-quantum-safe/oqs-provider/discussions/285

### CODE REVIEW

File: GOVERNANCE.md [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+
+Depending on the reason for removal, a Maintainer may be converted to Emeritus status. Emeritus Maintainers may still be consulted on some project matters, and can be returned to Maintainer status if their availability changes and a simple majority of Committers agrees.
+
+## Voting
+
+Change of role or changes to this document is subject to voting.
+
+Votes are to be executed by way of open github discussions. No quorum is needed for votes open for 4 weeks. Urgent matters may be decided by majority vote among Maintainers or 2/3 majority by all Committers within an arbitrary voting period.
+
+Any Committer may voluntarily temporarily for a documented period step down from the role losing voting rights for the specified period. If such period extends beyond one year, the Committer permanently loses role and rights. The period is documented in this file next to the person's name below.
```
Comment (SWilson4) [inline GOVERNANCE.md:45]: What is meant by "permanently" here? As written, it sounds like someone who steps down from the Committer role for a year is banned from regaining that status in the future. That seems heavy-handed to me. If the intent is more lenient (e.g., "permanently" means restarting from User), then I think this needs to be reworded.

**Review by SWilson4 — COMMENTED** [review]
Thanks very much for taking the time to do this @baentsch! I realize this is still marked as draft, so I hope I'm not stepping your toes by giving feedback now (since it was solicited in the latest meeting).

File: GOVERNANCE.md (hunk 1 above)
Comment (baentsch) [reply GOVERNANCE.md]: Resolved via #297

File: GOVERNANCE.md [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+
+Any person violating the [code of conduct](CODE_OF_CONDUCT.md), consistently not fulfilling the role responsibilities or other reasons can lose the role held if a simple majority of Committers votes for such removal and no Maintainer vetos that decision. If a Maintainer is to be removed from that role a 2/3 majority of Committers must agree.
+
+Depending on the reason for removal, a Maintainer may be converted to Emeritus status. Emeritus Maintainers may still be consulted on some project matters, and can be returned to Maintainer status if their availability changes and a simple majority of Committers agrees.
+
+## Voting
+
+Change of role or changes to this document is subject to voting.
+
+Votes are to be executed by way of open github discussions. No quorum is needed for votes open for 4 weeks. Urgent matters may be decided by majority vote among Maintainers or 2/3 majority by all Committers within an arbitrary voting period.
```
[the remark this answers is not in the document]
Comment (baentsch) [reply GOVERNANCE.md]: Resolved via #297

File: GOVERNANCE.md [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+
+Any User may become Contributor by creating a pull request (PR) and getting it successfully reviewed and merged by Committers.
+
+Any Contributor can become Committer by contributing sufficient code and displaying deep subject matter knowledge in discussions such that a majority of Committers vote for this change of role. A Maintainer can veto such vote. Such veto can be overruled by a 2/3 majority of Committers.
+
+As such voting decision may be considered subjective, Contributors striving to become Committers are encouraged to ask for advice by Committers as to what --if anything-- should be done (additionally to already documented knowledge in contributions) to attain this status. Baseline requirements for contributions are documented in [CONTRIBUTING.md](CONTRIBUTING.md). Any Contributor can create a discussion item to request a vote to become Committer.
+
+Any Committer can become Maintainer by majority vote of voting Committers. A current Maintainer can veto such vote. Such veto can be overruled by a 2/3 majority of all Committers.
+
+A Maintainer is not permitted to remove another Maintainer's github privileges.
```
[the remark this answers is not in the document]
Comment (baentsch) [reply GOVERNANCE.md]: Resolved via #297

File: GOVERNANCE.md [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+
+Any User may also be Contributor. Any Contributor may also be Committer. Any Committer may also be Maintainer. A Maintainer must be a Committer.
+
+## Change of role
+
+Any User may become Contributor by creating a pull request (PR) and getting it successfully reviewed and merged by Committers.
+
+Any Contributor can become Committer by contributing sufficient code and displaying deep subject matter knowledge in discussions such that a majority of Committers vote for this change of role. A Maintainer can veto such vote. Such veto can be overruled by a 2/3 majority of Committers.
+
+As such voting decision may be considered subjective, Contributors striving to become Committers are encouraged to ask for advice by Committers as to what --if anything-- should be done (additionally to already documented knowledge in contributions) to attain this status. Baseline requirements for contributions are documented in [CONTRIBUTING.md](CONTRIBUTING.md). Any Contributor can create a discussion item to request a vote to become Committer.
```
[the remark this answers is not in the document]
Comment (baentsch) [reply GOVERNANCE.md]: To be resolved as part of LF takeover negotiations (@dstebila fyi)

File: GOVERNANCE.md [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+
+This file documents the governance guidelines used for this project.
+
+## Roles
+
+The following roles exist in the project:
+
+1. Maintainer: Person with github administrative rights.
+
+2. Committer: Person with github "Write" privileges; this entails the right and obligation to review PRs by Contributors and to actively participate in discussions.
```
[the remark this answers is not in the document]
Comment (baentsch) [reply GOVERNANCE.md]: Resolved via #297

File: GOVERNANCE.md [hunk 6]
```diff
@@ -0,0 +1,63 @@
+# Governance
+
+This file documents the governance guidelines used for this project.
+
+## Roles
+
+The following roles exist in the project:
+
+1. Maintainer: Person with github administrative rights.
```
[the remark this answers is not in the document]
Comment (baentsch) [reply GOVERNANCE.md]: Resolved via #297

File: .github/ISSUE_TEMPLATE/discussion.md [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+about: Ask question, request feature or documentation update
+---
+
+### Thank you for taking the time to share your thoughts with us.
+
+We welcome all suggestions for new features or documentation updates
+helping others to make better use of this project.
+
+As with any other topic you'd like to discuss with the community,
+we'd primarily like to invite you to [open a discussion](https://github.com/openssl/openssl/discussions)
```
[the remark this answers is not in the document]
Comment (baentsch) [reply .github/ISSUE_TEMPLATE/discussion.md]: Resolved via #297

File: CONTRIBUTING
Comment (dstebila) [inline CONTRIBUTING:1]: I don't know what this is trying to say, nor do I think it necessary yet.

Comment (baentsch) [reply CONTRIBUTING:1]: Have removed that reference. Will add again if/when LinuxFoundation contracts don't reference the file "GOVERNANCE.md" (as overriding their control rules).

File: GOVERNANCE.md [hunk 8]
```diff
@@ -0,0 +1,70 @@
...
+
+Depending on the reason for removal, a Maintainer may be converted to Emeritus status. Emeritus Maintainers may still be consulted on some project matters, and can be returned to Maintainer status if their availability changes and a simple majority of Committers agrees.
+
+## Leave of absence
+
+Any Committer may voluntarily step down from the role for a documented period of time, losing voting rights for that time period. The period is documented in this file next to the person's name below.  At the end of this time period, the Committer automatically regains their voting rights.
+
+A leave of absence may not be longer than a year.  If the Committer needs to be away for longer than that, they must step down from that role unconditionally, and regaining that role becomes subject of normal procedures to become Committer, as described in ["Change of role"](#change-of-role) above.
```
Comment (SWilson4) [inline GOVERNANCE.md:47]:
```suggestion
A leave of absence may not be longer than a year. If the Committer needs to be away for longer than that, they must step down from that role unconditionally, and regaining that role becomes subject of normal procedures to become Committer, as described in ["Change of role"](#change-of-role) above.
```
Sorry for commenting on a triviality but... I can't not see the double space here :grimacing:

**Review by SWilson4 — APPROVED** [review]
I'm happy with this. Thanks @baentsch for bringing it together.

**Pull request merged**